cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js

```diff
@@ -162,7 +162,9 @@ class HydratorPlusPlusLeftPanelCtrl {
       let connections = [];
       let i;
       for (i=0; i<nodes.length - 1 ; i++) {
-        connections.push({ from: nodes[i].name, to: nodes[i+1].name });
+        if(nodes[i] && nodes[i + 1]) {
+          connections.push({ from: nodes[i].name, to: nodes[i+1].name });
```

Comment:
So if there are no nodes we won't enter the loop here right? Do we need this check? Also this is not including the fact that the pipeline can have pre-post run actions or am I missing something here?